src/Tests/Microsoft.Diagnostics.Monitoring.Tool.FunctionalTests/Runners/MonitorCollectRunner.CollectionRule.cs

```diff
@@ -38,19 +38,57 @@ public async Task WaitForCollectionRuleCompleteAsync(string ruleName, Cancellati
             }
         }
 
+        public async Task WaitForCollectionRuleUnmatchedFiltersAsync(string ruleName, CancellationToken token)
+        {
+            TaskCompletionSource<object> tcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            CollectionRuleKey key = new(LoggingEventIds.CollectionRuleUnmatchedFilters, ruleName);
+
+            AddCollectionRuleCallback(key, tcs);
+
+            try
+            {
+                await tcs.WithCancellation(token);
+            }
+            finally
+            {
+                RemoveCollectionRuleCallback(key, tcs);
+            }
+        }
+
+        public async Task WaitForCollectionRuleStartedAsync(string ruleName, CancellationToken token)
+        {
+            TaskCompletionSource<object> tcs = new(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            CollectionRuleKey key = new(LoggingEventIds.CollectionRuleStarted, ruleName);
+
+            AddCollectionRuleCallback(key, tcs);
+
+            try
+            {
+                await tcs.WithCancellation(token);
+            }
+            finally
+            {
+                RemoveCollectionRuleCallback(key, tcs);
+            }
+        }
+
         private void HandleCollectionRuleEvent(ConsoleLogEvent logEvent)
         {
             if (logEvent.State.TryGetValue("ruleName", out string ruleName))
             {
                 CollectionRuleKey key = new(logEvent.EventId, ruleName);
                 switch (logEvent.EventId)
                 {
-                    case LoggingEventIds.CollectionRuleFailed:
-                        FailCollectionRuleCallbacks(key, logEvent.Exception);
-                        break;
                     case LoggingEventIds.CollectionRuleCompleted:
+                    case LoggingEventIds.CollectionRuleUnmatchedFilters:
+                    case LoggingEventIds.CollectionRuleStarted:
                         CompleteCollectionRuleCallbacks(key);
                         break;
+                    case LoggingEventIds.CollectionRuleFailed:
+                        FailCollectionRuleCallbacks(key, logEvent.Exception);
+                        break;
                 }
```

Comment:
Would it make sense for us to have a default here that throws an `ArgumentException`? You would know better than me, but in the event that there was a different `LoggingEventId` added in the future, that would prevent it from just falling through this switch statement undetected if we forgot to update it.

Reply:
There a lot of other collection rule events that we aren't using in the tests. We would have to handle all of those here if we were to throw ArgumentException for unhandled events. So I don't think there's much value in doing that since we don't care about those events. The only reason why you'd want to add the event ID here is to have some kind of callback notification for that particular event, and we don't necessarily want to do that for all events unless we have to.